src/sonic-yang-models/yang-models/sonic-route-map.yang

```diff
@@ -0,0 +1,359 @@
+module sonic-route-map {
+    namespace "http://github.com/Azure/sonic-route-map";
+    prefix srmap;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix cmn;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-routing-policy-sets {
+        prefix srpolsets;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    import sonic-vrf {
+        prefix svrf;
+    }
+
+    import sonic-interface {
+        prefix sintf;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC Route map YANG";
+
+    revision 2021-02-26 {
+        description
+            "Initial revision.";
+    }
+
+    typedef route-map-intf {
+        type union {
+            type leafref {
```

Comment:
Please include subinterface too

Reply:
We'll add it once the subinterfaces YANG is available, not as part of this initial PR.